### ISSUE TITLE
Issues with Nested Workflows which take Collections as Input

### ISSUE THREAD

**Issue by alexlenail** [galaxyproject/galaxy#6569]
I have a workflow I'd like to assemble from two subworkflows. Let's call the top-level workflow A and the subworkflows B and C. 

B takes a collection of pairs of fastqs and returns a collection of bams. 
C takes a collection of bams and returns a collection of small files and plots. 

I'd like to build workflow A which takes a collection of fastqs and returns a collection of small files and plots. I'd like workflow A to be very simple: 
```
input collection -----> B -----> C -----> done. 
```
However, I can't drag a noodle from B to C. The noodle I drag from the input collection to B is also not a "multi"-noodle. It's clear the workflow UI thinks the output of B is a single file, though I know it isn't from running B individually. 

----

Bonus issue: 

When I add subworkflows to a workflow, e.g. in the workflow editor UI for A, I select B from the side panel, it appears in the workflow UI with its old name. That is, I have renamed that workflow, it shows up on the sidebar with its new name, but once I click it, a tool shows up in the editor UI with the old name. 

(cc @bgruening)

**Comment by mvdbeek** [comment]
The non-bonus part should work after #6732, the bonus part should be a separate issue with a simple example.
